zbench.zig

```diff
@@ -34,204 +35,217 @@ pub const Config = struct {
     /// Time budget for the benchmark in nanoseconds. Default is 2e9 (2 seconds).
     /// This value is used to determine how long a single benchmark should be allowed to run
     /// before concluding. Helps in avoiding long-running benchmarks.
-    time_budget: u64 = 2e9, // 2 seconds
+    time_budget_ns: u64 = 2e9,
 
     /// Configuration for lifecycle hooks in benchmarking.
     /// Provides the ability to define custom actions at different stages of the benchmark process:
-    /// - `beforeAll`: A hook that runs once before all benchmarks begin.
-    /// - `afterAll`: A hook that runs once after all benchmarks have completed.
-    /// - `beforeEach`: A hook that runs before each individual benchmark.
-    /// - `afterEach`: A hook that runs after each individual benchmark.
+    /// - `before_all`: A hook that runs once before all benchmarks begin.
+    /// - `after_all`: A hook that runs once after all benchmarks have completed.
+    /// - `before_each`: A hook that runs before each individual benchmark.
+    /// - `after_each`: A hook that runs after each individual benchmark.
     /// This allows for custom setup and teardown operations, as well as fine-grained control
     /// over the environment in which benchmarks are run.
-    hooks: LifecycleHooks = .{},
+    hooks: Hooks = .{},
+};
 
-    /// Flag to indicate whether system information should be displayed. Default is false.
-    /// If true, detailed system information (e.g., CPU, memory) will be displayed
-    /// along with the benchmark results. Useful for understanding the environment
-    /// in which the benchmarks were run.
-    display_system_info: bool = false,
+/// A benchmark definition.
+const Definition = struct {
+    name: []const u8,
+    func: BenchFunc,
+    config: Config,
 };
 
-/// Benchmark is a type representing a single benchmark session.
-/// It provides metrics and utilities for performance measurement.
-pub const Benchmark = struct {
-    /// Used to represent the 75th, 99th and 99.5th percentiles of the recorded durations,
-    /// generated by `Benchmark.calculatePercentiles`.
-    pub const Percentiles = struct {
-        p75: u64,
-        p99: u64,
-        p995: u64,
-    };
+/// A function pointer type that represents a benchmark function.
+pub const BenchFunc = *const fn (std.mem.Allocator) void;
```

Comment:
I was thinking of leaving the `Allocator` parameter for benchmark functions since it could be a relatively easy way of measuring memory use, make a allocator wrapper which records allocations and frees and pass that to the benchmark functions. It might need more thought than that but adding it now reduces API churn if it sounds like a workable idea.